fake.go

```diff
@@ -519,6 +521,177 @@ func (fake *Fake) Dump() string {
+		}
+		if words[1] != "table" && nft.Table == nil {
+			return fmt.Errorf("no table was added, can't add %s", words[1])
+		}
+		switch words[1] {
+		case "table":
+			if nft.Table != nil {
+				return fmt.Errorf("only 1 table should be declared")
+			}
+			tableObj := &FakeTable{
+				Chains: make(map[string]*FakeChain),
+				Sets:   make(map[string]*FakeSet),
+				Maps:   make(map[string]*FakeMap),
+			}
+			if len(words) > 4 {
+				_, comment := extractComment(strings.Join(words[4:], " "))
+				if comment != "" {
+					tableObj.Comment = &comment
+				}
+			}
+			nft.Table = tableObj
+		case "chain":
+			chainObj := Chain{
+				Name: words[4],
+			}
+			if len(words) > 5 {
+				chainStr, comment := extractComment(strings.Join(words[6:len(words)-1], " "))
+				if len(comment) > 0 {
+					chainObj.Comment = &comment
+				}
+				if len(chainStr) > 0 {
+					chainWords := strings.Split(chainStr, " ")
+					if len(chainWords) != 7 {
+						return fmt.Errorf("can't parse chain description: %s", chainStr)
+					}
+					chainType, _ := parseChainType(chainWords[1])
+					chainHook, _ := parseBaseChainHook(chainWords[3])
+					chainObj.Type = &chainType
+					chainObj.Hook = &chainHook
+					chainObj.Priority = (*BaseChainPriority)(&chainWords[5])
+				}
+			}
+			nft.Table.Chains[words[4]] = &FakeChain{
+				Chain: chainObj,
+			}
+		case "rule":
+			chainName := words[4]
+			rule, comment := extractComment(strings.Join(words[5:], " "))
+			ruleObj := &Rule{
+				Chain: chainName,
+				Rule:  rule,
+			}
+			if len(comment) > 0 {
+				ruleObj.Comment = &comment
+			}
+			nft.Table.Chains[chainName].Rules = append(nft.Table.Chains[chainName].Rules, ruleObj)
+		case "map":
+			mapName := words[4]
+			mapProps, comment := extractComment(strings.Join(words[6:len(words)-1], " "))
+			typeReg := regexp.MustCompile(`type(.*?);`)
+			typeSection := typeReg.FindStringSubmatch(mapProps)[0]
+			mapType := strings.TrimPrefix(typeSection, "type ")
+			mapType = strings.TrimSuffix(mapType, " ;")
+
+			mapObj := Map{
+				Name: mapName,
+				Type: mapType,
+			}
+			if len(comment) > 0 {
+				mapObj.Comment = &comment
+			}
+			nft.Table.Maps[mapName] = &FakeMap{
+				Map: mapObj,
+			}
+		case "set":
+			setName := words[4]
+			setStr, comment := extractComment(strings.Join(words[6:len(words)-1], " "))
+			setProps := strings.Split(setStr, " ")
+			flags, err := parseSetFlags(setProps[4])
+			if err != nil {
+				return err
+			}
+			timeout, err := time.ParseDuration(setProps[7])
+			if err != nil {
+				return err
+			}
+			setObj := Set{
+				Name:    setName,
+				Type:    setProps[1],
+				Flags:   flags,
+				Timeout: &timeout,
+			}
+			if len(comment) > 0 {
+				setObj.Comment = &comment
+			}
+			nft.Table.Sets[setName] = &FakeSet{
+				Set: setObj,
+			}
+		case "element":
+			elementObj := &Element{}
+			mapOrSetName := words[4]
+			if mapObj, ok := nft.Table.Maps[mapOrSetName]; ok {
```

Comment:
you can also figure this out purely syntactically, based on whether the element has just keys, or keys and values